runtime/include/cloe/data_broker.hpp

```diff
@@ -0,0 +1,1099 @@
+   *
+   * Example:
+   * ```
+   * Signal s = ...;
+   * s.set_getter<type>([&]() -> const type & { return value; });
+   * ```
+   * Usage Note: When using lambdas, the explicit return-type definition is important!
+   *
+   * Undefined behaviour:
+   * ```
+   * Signal s = ...;
+   * s.set_getter<type>([&]()     { return value; });
+   *                          ^^^ No return type specified, type conversion rules apply
+   * ```
+   */
+  template <typename T>
+  void set_getter(typed_get_value_function_t<T> get_value_fn) {
+    assert_static_type<T>();
+    assert_dynamic_type<T>();
+
+    get_value_ = std::move(get_value_fn);
+  }
+
+  /**
+   * Return the current value of the signal.
+   *
+   * \tparam T Type of the signal
+   * \return databroker::signal_type_cref_t<T>, Current value of the signal
+   * \note databroker::compatible_base_t<T> == T, if the method compiles
+   */
+  template <typename T>
+  databroker::signal_type_cref_t<T> value() const {
+    assert_static_type<T>();
+    assert_dynamic_type<T>();
+    using compatible_type = databroker::compatible_base_t<T>;
+
+    const typed_get_value_function_t<T>* getter_fn = getter<compatible_type>();
+    if (getter_fn && (*getter_fn)) {
+      databroker::signal_type_cref_t<T> value = getter_fn->operator()();
+      return value;
+    }
+    throw std::logic_error(
+        fmt::format("unable to get value for signal without getter-function: {}", names_.front()));
+  }
+
+  /**
+   * Return the getter function of the signal.
+   *
+   * \tparam T Type of the signal
+   * \return const typed_set_value_function_t<T>*, Getter function of the signal
+   */
+  template <typename T>
+  const typed_set_value_function_t<T>* setter() const {
+    assert_static_type<T>();
+    assert_dynamic_type<T>();
+
+    const typed_set_value_function_t<T>* set_value_fn =
+        std::any_cast<typed_set_value_function_t<T>>(&set_value_);
+    return set_value_fn;
+  }
+
+  /**
+   * Set the setter function of the signal.
+   *
+   * \tparam T Type of the signal
+   * \param set_value Getter function of the signal
+   */
+  template <typename T>
+  void set_setter(typed_set_value_function_t<T> set_value_fn) {
+    assert_static_type<T>();
+    assert_dynamic_type<T>();
+
+    set_value_ = std::move(set_value_fn);
+  }
+
+  /**
+   * Set the value of the signal.
+   *
+   * \tparam T Type of the signal
+   * \param value Value of the signal
+   */
+  template <typename T>
+  void set_value(databroker::signal_type_cref_t<T> value) const {
+    assert_static_type<T>();
+    assert_dynamic_type<T>();
+    using compatible_type = databroker::compatible_base_t<T>;
+
+    const typed_set_value_function_t<T>* setter_fn = setter<compatible_type>();
+    if (setter_fn && *setter_fn) {
+      setter_fn->operator()(value);
+      return;
+    }
+    throw std::logic_error(
+        fmt::format("unable to set value for signal without setter-function: {}", names_.front()));
```

Comment:
According to the rest of the class, `names_` can be empty. So we should check here.
But I am also led to believe that a signal with no name is rather pointless.

Help me understand the use-case for a signal with no name.
The only use case I can think of is one where a name is supplied later, but I'm not sure if we need this.